src/nspawn/nspawn.c

```diff
@@ -2162,6 +2163,57 @@ static int setup_boot_id(void) {
         return mount_nofollow_verbose(LOG_ERR, NULL, to, NULL, MS_BIND|MS_REMOUNT|MS_RDONLY|MS_NOSUID|MS_NOEXEC|MS_NODEV, NULL);
 }
 
+static int get_fuse_version(uint32_t *major, uint32_t *minor) {
+        int r;
+        ssize_t n;
+        _cleanup_close_ int fd;
+        _cleanup_(umount_and_rmdir_and_freep) char *dir;
+        _cleanup_free_ char *opts = NULL;
+        union {
+                char unstructured[FUSE_MIN_READ_BUFFER];
+                struct {
+                        struct fuse_in_header header;
+                        /* Don't use <linux/fuse.h>:`struct fuse_init_in` because a newer fuse.h might give
+                         * us a bigger struct than what an older kernel actually gives us, and that would
+                         * break our short-read checks. */
+                        struct {
+                                uint32_t major;
+                                uint32_t minor;
+                        } body;
+                } structured;
+        } request;
+
+        /* Get a FUSE handle. */
+        fd = open("/dev/fuse", O_RDWR);
+        if (fd < 0)
+                return log_warning_errno(errno, "Disabling FUSE: Failed to open /dev/fuse: %m");
+        r = mkdtemp_malloc("/tmp/nspawn-fuse-XXXXXX", &dir);
+        if (r < 0)
+                return log_warning_errno(r, "Disabling FUSE: Failed to create temporary directory: %m");
+        if (asprintf(&opts, "fd=%i,rootmode=40000,user_id=0,group_id=0", fd) < 0)
+                return log_oom();
+        if (mount("nspawn-fuse", dir, "fuse.nspawn", 0, opts) < 0)
```

Comment:
hmm, this is in the host mount namespace? this would create a mount in the host? please don't, that would trigger all kind of a mess, because people watch mount tables, including systemd itself.

can't you just use the new mount apis? this allows you to create an anonymous superblock without havint to attach it anywhere. that should suffice, no?

Reply:
This is not in the host namespace (I'll add a comment calling this out); indeed that would cause a mess.

Isn't `fsopen()` much newer than what we're doing a version check for anyway? If I could assume `fsopen()` works, then couldn't I just assume that FUSE is new enough?

Reply:
you never know what people backport and what they don't.

given that this is a new feature of nspawn it doesn't concern us much if fuse already supported this before fsopen() was added. we are not breaking anything if we require both fsopen() and new fuse here.

Reply:
Done.